I'm not sure if this package even supports darwin (I can't find it mentioned anywhere upstream).

These logs seem to indicate the `tree_sitter_devicetree` module (a transitive dependency) is trying to import a `parser` library, and cypthon can't find it:

```
Did not find version 0.22.0 in the output of the command /nix/store/1p0c95n6272qh059q14qrc6chpvcwv6x-python3.13-keymap-drawer-0.22.0/bin/keymap --version

Traceback (most recent call last):
  File "{{storeDir}}/bin/.keymap-wrapped", line 6, in <module>
    from keymap_drawer.__main__ import main
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/__main__.py", line 17, in <module>
    from keymap_drawer.draw import KeymapDrawer
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/draw/__init__.py", line 3, in <module>
    from .draw import KeymapDrawer
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/draw/draw.py", line 13, in <module>
    from keymap_drawer.draw.combo import ComboDrawerMixin
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/draw/combo.py", line 8, in <module>
    from keymap_drawer.draw.utils import UtilsMixin
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/draw/utils.py", line 11, in <module>
    from keymap_drawer.draw.glyph import GlyphMixin
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/draw/glyph.py", line 20, in <module>
    from keymap_drawer.keymap import KeymapData, LayoutKey
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/keymap.py", line 14, in <module>
    from keymap_drawer.physical_layout import PhysicalLayout, PhysicalLayoutGenerator
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/physical_layout.py", line 24, in <module>
    from keymap_drawer.dts import DeviceTree
  File "{{storeDir}}/lib/python3.13/site-packages/keymap_drawer/dts.py", line 16, in <module>
    import tree_sitter_devicetree as ts
  File "{{storeDir}}/lib/python3.13/site-packages/tree_sitter_devicetree/__init__.py", line 1, in <module>
    from ._binding import language
ImportError: dlopen({{storeDir}}/lib/python3.13/site-packages/tree_sitter_devicetree/_binding.cpython-313-darwin.so, 0x0002): Library not loaded: parser
  Referenced from: <F0D5A9F8-FECD-3275-A29C-121AABBC74EC> {{storeDir}}/lib/python3.13/site-packages/tree_sitter_devicetree/_binding.cpython-313-darwin.so
  Reason: tried: 'parser' (no such file), '/usr/local/lib/parser' (no such file), '/usr/lib/parser' (no such file, not in dyld cache), '//parser' (no such file), '/usr/local/lib/parser' (no such file), '/usr/lib/parser' (no such file, not in dyld cache)

error: Cannot build '/nix/store/wcw17f8sk7anm3wpzhdji71xpf8v1lc7-python3.13-keymap-drawer-0.22.0.drv'.
```

This gets caught by the `versionCheckHook` when it tries to run `keymap --version`.

I can mark it as `broken` on darwin, or I can specify `platforms` as linux[^1], or I can investigate possible solutions (if anyone has ideas).

[^1]: or use `buildPythonModule`'s `disabled` attr on darwin?

--- a/pkgs/by-name/ke/keymap-drawer/package.nix
+++ b/pkgs/by-name/ke/keymap-drawer/package.nix
@@ -1,0 +1,4 @@
+{
+  python3Packages,
+}:
+python3Packages.toPythonApplication python3Packages.keymap-drawer

--- a/pkgs/development/python-modules/keymap-drawer/default.nix
+++ b/pkgs/development/python-modules/keymap-drawer/default.nix
@@ -1,0 +1,69 @@
+{
+  lib,
+
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+
+  nix-update-script,
+  pcpp,
+  platformdirs,
+  poetry-core,
+  pydantic,
+  pydantic-settings,
+  pyparsing,
+  pyyaml,
+  tree-sitter,
+  tree-sitter-grammars,
+  versionCheckHook,
+}:
+let
+  version = "0.22.0";
+in
+buildPythonPackage {
+  inherit version;
+  pname = "keymap-drawer";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    owner = "caksoylar";
+    repo = "keymap-drawer";
+    tag = "v${version}";
+    hash = "sha256-SPnIfrUA0M9xznjEe60T+0VHh9lCmY4cni9hyqFlZqM=";
+  };
+
+  build-system = [ poetry-core ];
+
+  dependencies = [
+    pcpp
+    platformdirs
+    pydantic
+    pydantic-settings
+    pyparsing
+    pyyaml
+    tree-sitter
+    tree-sitter-grammars.tree-sitter-devicetree
+  ];
+
+  nativeCheckInputs = [
+    versionCheckHook
+  ];
+
+  pythonImportsCheck = [ "keymap_drawer" ];
+
+  versionCheckProgram = "${placeholder "out"}/bin/keymap";
+  versionCheckProgramArg = "--version";
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Module and CLI tool to help parse and draw keyboard layouts";
+    homepage = "https://github.com/caksoylar/keymap-drawer";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [
+      MattSturgeon
+    ];
+    mainProgram = "keymap";
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7584,6 +7584,8 @@ self: super: with self; {
 
   keyboard = callPackage ../development/python-modules/keyboard { };
 
+  keymap-drawer = callPackage ../development/python-modules/keymap-drawer { };
+
   keyring = callPackage ../development/python-modules/keyring { };
 
   keyring-pass = callPackage ../development/python-modules/keyring-pass { };

